jans-cedarling/bindings/cedarling_python/PYTHON_TYPES.md

```diff
@@ -5,98 +5,35 @@ This document describes the Cedarling Python bindings types.
-:param id_token_trust_mode: Trust mode for ID tokens, either 'None' or 'Strict'.  
-:param lock: Enables integration with Lock Master for policies and SSE events.  
-:param lock_master_configuration_uri: URI where Cedarling can get JSON file with all required metadata about Lock Master, i.e. .well-known/lock-master-configuration.  
-:param dynamic_configuration: Toggles listening for SSE config updates.  
-:param lock_ssa_jwt: SSA for DCR in a Lock Master deployment. Cedarling will validate this SSA JWT prior to DCR.  
-:param audit_log_interval: Interval (in seconds) for sending log messages to Lock Master (0 to disable).  
-:param audit_health_interval: Interval (in seconds) for sending health updates to Lock Master (0 to disable).  
-:param audit_health_telemetry_interval: Interval (in seconds) for sending telemetry updates to Lock Master (0 to disable).  
-:param listen_sse: Toggles listening for updates from the Lock Server.
+A Python wrapper for the Rust `cedarling::BootstrapConfig` struct.
+Configures the `Cedarling` application, including authorization, logging, and policy store settings.
 
-Example
+Methods
 -------
-```python
-from cedarling import BootstrapConfig
-# Example configuration
-bootstrap_config = BootstrapConfig({
-    "application_name": "MyApp",
-    "policy_store_uri": None,
-    "policy_store_id": "policy123",
-    "log_type": "memory",
-    "log_ttl": 86400,
-    "decision_log_user_claims": ["sub", "email", "username"]
-    "decision_log_workload_claims": ["client_id", "rp_id"]
-    "decision_log_default_jwt_id":"jti"
-    "user_authz": "enabled",
-    "workload_authz": "enabled",
-    "usr_workload_bool_op": "AND",
-    "local_jwks": "./path/to/your_jwks.json",
-    "local_policy_store": None,
-    "policy_store_local_fn": "./path/to/your_policy_store.json",
-    "jwt_sig_validation": "enabled",
-    "jwt_status_validation": "disabled",
-    "at_iss_validation": "enabled",
-    "at_jti_validation": "enabled",
-    "at_nbf_validation": "disabled",
-    "idt_iss_validation": "enabled",
-    "idt_sub_validation": "enabled",
-    "idt_exp_validation": "enabled",
-    "idt_iat_validation": "enabled",
-    "idt_aud_validation": "enabled",
-    "userinfo_iss_validation": "enabled",
-    "userinfo_sub_validation": "enabled",
-    "userinfo_aud_validation": "enabled",
-    "userinfo_exp_validation": "enabled",
-    "id_token_trust_mode": "Strict",
-    "lock": "disabled",
-    "lock_master_configuration_uri": None,
-    "dynamic_configuration": "disabled",
-    "lock_ssa_jwt": None,
-    "audit_log_interval": 0,
-    "audit_health_interval": 0,
-    "audit_health_telemetry_interval": 0,
-    "listen_sse": "disabled",
-})
-```
+.. method:: __init__(self, config)
+
+    Initializes the Cedarling instance with the provided configuration.
+
+    :param options: A `dict` with startup settings.
+
+.. method:: load_from_file(str) -> BootstrapConfig
```

Comment:
I don't understand the approach of having a separate method to load from file vs loading from string. Isn't this the same as allowing the developer to read a file and then passing the string to `load_from_str`?

Reply:
Rust code has implementation to read from file. So it cost nothing to provide binding to it.

Reply:
Logging is not related to this PR

Logging will be fixed in the https://github.com/JanssenProject/jans/issues/10302

Reply:
`Yes, pyi file is what is loaded by Python editors and LSP, so parameters are essential to help developers.`
in this PR we're going to use only one parameter with dict type. 
so actually dict parameters is not part of LSP

Reply:
Ok then, as long as we provide info to developer on how to construct bootstrapconfig dict, fine by me